CMakeLists.txt

```
@@ -19,8 +19,11 @@ if(NOT TTLS_NOTEST)
 
   set(CMAKE_CXX_CLANG_TIDY clang-tidy-10)
 
-  if(CMAKE_BUILD_TYPE STREQUAL Debug)
-    add_compile_options(-fsanitize=thread)
+  if(TTLS_ASAN)
+    add_compile_options(-fsanitize=address,undefined)
+    add_link_options(-fsanitize=address,undefined)
+  elseif(TTLS_TSAN OR CMAKE_BUILD_TYPE STREQUAL Debug)
+    add_compile_options(-fsanitize=thread -fno-sanitize-recover=all)
```

```suggestion
    add_compile_options(-fsanitize=address,undefined -fno-sanitize-recover=all)
    add_link_options(-fsanitize=address,undefined)
  elseif(TTLS_TSAN OR CMAKE_BUILD_TYPE STREQUAL Debug)
    add_compile_options(-fsanitize=thread)
```
no-sanitize-recover is needed for UBSAN (only) because otherwise it just prints a warning and continues and the test does not fail.